modules/kafka/kafka-dest-driver.c

```diff
@@ -310,8 +305,8 @@ _kafka_delivery_report_cb(rd_kafka_t *rk,
                 evt_tag_str("error", rd_kafka_err2str(err)),
                 evt_tag_str("driver", self->super.super.super.id),
                 log_pipe_location_tag(&self->super.super.super.super));
-      log_msg_unref(msg);
     }
+  log_msg_unref(msg);
   log_threaded_dest_worker_wakeup_when_suspended((LogThreadedDestWorker *) self->super.workers[0]);
```

Comment:
just a note, I'll have to check it too:
I think wakeup here is not needed: the actual ack is done right after `insert`, I think wakeup here was meant for the error-handling case.

Reply:
Not really, but actually passing the whole logmsg probably not needed here at all.

Reply:
I agree, we can remove the LogMessage passing.

About the `log_threaded_dest_worker_wakeup_when_suspended`: I've double checked: this wakeup is not necessary anymore. It's purpose is to avoid the worker thread getting stuck (i.e. being idle while there are messages in its queue: when there is no new message, but we pushed back a failed-to-deliver message from the delivery_callback). However sometimes it didn't work: `log_threaded_dest_worker_wakeup_when_suspended` needed that the worker's state is `suspended = TRUE`, which only happens if the previous publish call in worker#0 failed.

Also, during shutdown the last flush attempt from deinit can cause a segmentation fault: as `log_threaded_dest_worker_wakeup_when_suspended` is being called from the main thread, which calls the `lib/logthrdest/logthrdestdrv.c:_perform_work` function, and that's leading that the main thread tries to access a different ivykis main loop's timers and objects (especially that the worker thread is already stopped, and the worker thread's ivykis main loop is already deinitialized).

Reply:
With this version I don't see the segfault anymore.

Also, on master branch, I've removed only `log_threaded_dest_worker_wakeup_when_suspended` for a test run to see if it fixes the crash and I didn't see any crash at all.